grpc_server/lib/grpc/server/cache.ex

```diff
@@ -0,0 +1,90 @@
+defmodule GRPC.Server.Cache do
+  @moduledoc """
+  ETS-based cache for frequently accessed server metadata.
```

Comment:
How frequently is this being written to? Does it make sense to use :persistent_term for better read performance?